### PR TITLE
Access control: Add metrics

### PR DESCRIPTION
**Pull request by donomii**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/1247

**Special notes for your reviewer**:

Branched from main and cherry-picked patches

### CODE REVIEW

File: pkg/services/accesscontrol/evaluator/evaluator.go [hunk 1]
```diff
@@ -5,13 +5,18 @@ import (
...
+	"github.com/grafana/grafana/pkg/infra/metrics"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 // Evaluate evaluates access to the given resource, using provided AccessControl instance.
 // Scopes are evaluated with an `OR` relationship.
 func Evaluate(ctx context.Context, ac accesscontrol.AccessControl, user *models.SignedInUser, action string, scope ...string) (bool, error) {
+	timer := prometheus.NewTimer(metrics.MAccessPermissionsSummary)
```
Comment (sakjur) [inline pkg/services/accesscontrol/evaluator/evaluator.go:17]: Won't this be registered twice per operation if we do it both in Evaluate and GetUserPermissions?

**Review by sakjur — COMMENTED** [review]
If you have time for it, in general I'd suggest having the counters live inside of the `pkg/services/accesscontrol` domain somewhere (say `pkg/services/accesscontrol/metrics` or even in the root) since we're moving more into package by service rather than package by layer.

That said, this is also good. I asked a question about possible double reporting that I'd like an answer to before approving, but that's the only thing 🙂

File: pkg/services/accesscontrol/evaluator/evaluator.go (hunk 1 above)
Comment (donomii) [reply pkg/services/accesscontrol/evaluator/evaluator.go:17]: I think this use is ok.  NewTimer doesn't register a metric, it does the measurement and writes it to metrics.MAccessPermissionsSummary, which is registered elsewhere.  timer is local to this call and is destroyed after every call, MAccessPermissionsSummary is the persistent metric. 

I hope.

**Comment by donomii** [comment]
> If you have time for it, in general I'd suggest having the counters live inside of the `pkg/services/accesscontrol` domain somewhere (say `pkg/services/accesscontrol/metrics` or even in the root) since we're moving more into package by service rather than package by layer.

Happy to move it, I'll make another issue for it, I'm out of time today

**Pull request closed without merging**